nautobot/extras/tests/test_models.py

```diff
@@ -1072,6 +1074,58 @@ def test_remote_url_hostname(self):
         self.repo.remote_url = "http://some-private-host/example.git"
         self.repo.validated_save()
 
+    def test_clone_context_manager(self):
+        """Confirm that the clone() context manager methods work as expected."""
+        self.tempdir = tempfile.TemporaryDirectory()  # pylint: disable=consider-using-with
+        create_and_populate_git_repository(self.tempdir.name)
+
+        self.repo_slug = "new_git_repo"
+        self.repo = GitRepository(
+            name="New Git Repository",
+            slug=self.repo_slug,
+            remote_url="file://" + self.tempdir.name,  # file:// URLs aren't permitted normally, but very useful here!
+            branch="main",
+            # Provide everything we know we can provide
+            provided_contents=[entry.content_identifier for entry in get_datasource_contents("extras.gitrepository")],
+        )
+        self.repo.save()
+        specified_path = tempfile.mkdtemp()
+        with self.subTest("Clone a repository with no path argument provided"):
+            with self.repo.clone_to_directory() as path:
+                # assert that the temporary directory was created in the expected location i.e. /tmp/
+                self.assertTrue(path.startswith(tempfile.gettempdir()))
+                self.assertTrue(os.path.exists(path))
```

Comment:
Would be good to add a spot-check or two to assert that the files in the repo are present in `path` and that the correct HEAD is checked out.